napari/view_layers.py

```diff
@@ -263,126 +263,108 @@ def imshow(
-        that is being expanded and then each colormap is applied to each
-        image.
-    gamma : list, float
-        Gamma correction for determining colormap linearity. Defaults to 1.
-        If a list then must be same length as the axis that is being
-        expanded as channels.
-    interpolation : str or list
-        Deprecated, to be removed in 0.6.0
-    interpolation2d : str or list
-        Interpolation mode used by vispy in 2D. Must be one of our supported
-        modes. If a list then must be same length as the axis that is being
-        expanded as channels.
-    interpolation3d : str or list
-        Interpolation mode used by vispy in 3D. Must be one of our supported
-        modes. If a list then must be same length as the axis that is being
-        expanded as channels.
-    rendering : str or list
-        Rendering mode used by vispy. Must be one of our supported
-        modes. If a list then must be same length as the axis that is being
-        expanded as channels.
-    depiction : str
-        Selects a preset volume depiction mode in vispy
-
-        * volume: images are rendered as 3D volumes.
-        * plane: images are rendered as 2D planes embedded in 3D.
-    iso_threshold : float or list
-        Threshold for isosurface. If a list then must be same length as the
-        axis that is being expanded as channels.
-    attenuation : float or list
-        Attenuation rate for attenuated maximum intensity projection. If a
-        list then must be same length as the axis that is being expanded as
-        channels.
-    name : str or list of str
-        Name of the layer.  If a list then must be same length as the axis
-        that is being expanded as channels.
-    metadata : dict or list of dict
-        Layer metadata. If a list then must be a list of dicts with the
-        same length as the axis that is being expanded as channels.
-    scale : tuple of float or list
-        Scale factors for the layer. If a list then must be a list of
-        tuples of float with the same length as the axis that is being
-        expanded as channels.
-    translate : tuple of float or list
-        Translation values for the layer. If a list then must be a list of
-        tuples of float with the same length as the axis that is being
-        expanded as channels.
-    rotate : float, 3-tuple of float, n-D array or list.
-        If a float convert into a 2D rotation matrix using that value as an
-        angle. If 3-tuple convert into a 3D rotation matrix, using a yaw,
-        pitch, roll convention. Otherwise assume an nD rotation. Angles are
-        assumed to be in degrees. They can be converted from radians with
-        np.degrees if needed. If a list then must have same length as
+        All parameters except data, rgb, and multiscale can be provided as
+        list of values. If a list is provided, it must be the same length as
```

Comment:
I think this can be a bit confusing for people as data can be provided as a list of values in which case it is assumed to be multiscale.

Reply:
Just checked and confirmed that data can be provided as a list of arrays which can be counted also when `channel_axis` is provided.

Reply:
> I think this can be a bit confusing for people as data can be provided as a list of values in which case it is assumed to be multiscale.

Not sure how to better handle this. Here we tried to be explicit: data docstring says how lists of data are handled: multiscale case. channel_axis says how the other parameter lists are handled. 


> Just checked and confirmed that data can be provided as a list of arrays which can be counted also when channel_axis is provided.

Can you clarify what you did?
If I make 2 np array say (100, 100) and pass them as list I cannot use channel_axis:
`ValueError: Input data should be an array-like object, or a sequence of arrays of decreasing size. Got arrays of single size: 10000`

Reply:
I responded to @psobolewskiPhD over zulip but just for completeness here:
```
import napari
from skimage.data import astronaut

# last dimension is the one containing the rgb channels
data=astronaut()
viewer = napari.Viewer()

viewer.add_image(data=[data, data[::2,::2,:], data[::4,::4, :]], channel_axis=2)
```

Reply:
Right, but that's not an example of channel_axis mode working with data.
The data list is not being treated as 3 layers, instead  because data is decreasing in size, it's detected as multiscale, it's treated as a single layer and then split as RGB.